### PR TITLE
Grpcio additions

### PR DESCRIPTION
**Pull request by happysalada**
###### Description of changes

small grpcio additions in preparation of bentoml

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/grpcio-health-checking/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  pythonImportsCheck = [ "grpc_health" ];
+
+  # no tests
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Standard Health Checking Service for gRPC";
+    homepage = "https://pypi.org/project/grpcio-health-checking/";
+    license = with licenses; [ ];
+    maintainers = with maintainers; [ ];
```
Comment (natsukium) [inline pkgs/development/python-modules/grpcio-health-checking/default.nix:40]:
```suggestion
    license = with licenses; [ asl20 ];
    maintainers = with maintainers; [ happysalada ];
```

Comment (happysalada) [reply pkgs/development/python-modules/grpcio-health-checking/default.nix]: Thanks for catching this!

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -4513,8 +4513,14 @@ self: super: with self; {
 
   grpcio = callPackage ../development/python-modules/grpcio { };
 
+  grpcio-channelz = callPackage ../development/python-modules/grpcio-channelz { };
+
+  grpcio-health-checking = callPackage ../development/python-modules/grpcio-health-checking { };
+
   grpcio-gcp = callPackage ../development/python-modules/grpcio-gcp { };
```
Comment (natsukium) [inline pkgs/top-level/python-packages.nix:4520]:
```suggestion
  grpcio-gcp = callPackage ../development/python-modules/grpcio-gcp { };

  grpcio-health-checking = callPackage ../development/python-modules/grpcio-health-checking { };
```
nit
Besides this, it looks good to me.

**Review by natsukium — APPROVED** [review]
LGTM, thanks!

**Comment by natsukium** [comment]
I forgot to post this.

Result of `nixpkgs-review pr 245001` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>python310Packages.grpcio-channelz</li>
    <li>python310Packages.grpcio-channelz.dist</li>
    <li>python310Packages.grpcio-health-checking</li>
    <li>python310Packages.grpcio-health-checking.dist</li>
    <li>python310Packages.grpcio-reflection</li>
    <li>python310Packages.grpcio-reflection.dist</li>
    <li>python311Packages.grpcio-channelz</li>
    <li>python311Packages.grpcio-channelz.dist</li>
    <li>python311Packages.grpcio-health-checking</li>
    <li>python311Packages.grpcio-health-checking.dist</li>
    <li>python311Packages.grpcio-reflection</li>
    <li>python311Packages.grpcio-reflection.dist</li>
  </ul>
</details>

**Pull request merged**